nixos/modules/programs/pazi.nix

```diff
@@ -0,0 +1,30 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.pazi;
+in
+{
+  options = {
+    programs.pazi = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable pazi.
+        '';
+      };
+    };
+  };
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ pkgs.pazi ];
+
+    programs.bash.interactiveShellInit = "source ${pkgs.pazi}/share/pazi.bash";
+    programs.fish.interactiveShellInit = "source ${pkgs.pazi}/share/pazi.fish";
+    programs.zsh.interactiveShellInit = "source ${pkgs.pazi}/share/pazi.zsh";
```

Comment:
I'd much rather have some conventions for where to put stuff that should be auto-sourced. Then people could just do `systemPackages = [ pkgs.pazi ]` and their shell would load it because pazi.* appears in some /run/current-system/sw/share/* path. This would then also work for `nix-env` and on non-NixOS systems.